### ISSUE TITLE
ValueError: mutable default <class 'pytest_tui.utils.TuiSection'> for field live_log_sessionstart is not allowed: use default_factory

### ISSUE THREAD

**Issue by matthewdeanmartin** [jeffwright13/pytest-tui#93]
Tried this out and it blew up.

pytest-tui 1.7.0
Python 3.11
Debian flavored Linux
Built-in terminal in VSCode

```
  File "/home/vscode/.local/share/virtualenvs/template_management_service-R81hyB01/lib/python3.11/site-packages/pytest_tui/utils.py", line 192, in <module>
    @dataclass
     ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1220, in dataclass
    return wrap(cls)
           ^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 1210, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 958, in _process_class
    cls_fields.append(_get_field(cls, name, type, kw_only))
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/dataclasses.py", line 815, in _get_field
    raise ValueError(f'mutable default {type(f.default)} for field '
ValueError: mutable default <class 'pytest_tui.utils.TuiSection'> for field live_log_sessionstart is not allowed: use default_factory
```

**Comment by jeffwright13** [comment]
Thanks for reporting this! Python 3.11 has some stricter rules around the use of dataclasses that I have not yet fixed in my code. I actually have [an issue](https://github.com/jeffwright13/pytest-tui/issues/84) filed against the repo to fix this, just didn't get around to it yet - I'll work on it this weekend. In the meantime, can you re-try with Python 3.8, 3.9 or 3.10?

**Comment by jeffwright13** [comment]
I've coded up a solution and posted the new package at https://test.pypi.org/project/pytest-tui/. Please give it a try and let me know if it solves your problem. I am still testing it, which is why I have not yet released it officially.

**Comment by matthewdeanmartin** [comment]
Works for python3.11, ubuntu, gnome terminal. 

However

```bash
pytest --tui
```
has no effect, is it supposed to do anything?

```bash
tui
```
does launch a terminal ui.

more info

```
pytest==7.2.1
pytest-cov==4.0.0
pytest-metadata==2.0.4
pytest-tui @ file:///home/mmartin/github/dedlin/pytest_tui-1.7.2-py2.py3-none-any.whl
```

**Comment by jeffwright13** [comment]
`pytest --tui` simply runs the Pytest session, and saves the test results in a format that either of the two console scripts (`tui` | `tuih`) can use to generate the results via their interfaces. So it's a two-step process right now. I kind of liken it to `pytest-html`, where you give the `--html=report.html` flag to the Pytest run, then you go find the results file and launch it in a browser.

At one point I had a whole configuration infrastructure built around `tui` and `tuih`, where you could set the scripts to auto-launch after running the Pytest session. I retired it because it felt intrusive, and it was a hassle to maintain. I've been considering bringing it back again, but before I do that, I need to focus on getting a thorough set of automated tests in place that will tell me if I've broken something during development. There have been too many times when I've made a seemingly innocent change, only to find out something was not working anymore.

Thanks for verifying the fix. I'll go ahead and get it up on Pypi.org soon.

**Comment by jeffwright13** [comment]
Fixed in 1.8.0
